File: Sliding Window/2294. Partition Array Such That Maximum Difference Is K/Solution.java
class Solution {
    public int partitionArray(int[] nums, int k) {
        // Time Complexity: O(NlogN)
        // Space Complexity: O(1)
        Arrays.sort(nums);
        int n = nums.length;
        int l = 0;
        int r = 0;
        int groups = 1;
        
        while (r < n){
            if (nums[r] - nums[l] > k){
                groups++;
                l = r;
            }
            r++;
        }
        return groups;
    }
}

File: Sliding Window/2294. Partition Array Such That Maximum Difference Is K/Solution.py
from typing import List

class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        # Time Complexity: O(NlogN)
        # Space Complexity: O(1)
        nums.sort()
        n = len(nums)
        l, r = 0, 0
        groups = 1
        while r < n:
            if nums[r] - nums[l] > k:
                groups += 1
                l = r
            r += 1
        return groups